### DISCUSSION TITLE
Tips on compiling MAC projects using mvn and JDK > 8

### DISCUSSION THREAD

**Discussion by alick888** [MuleSoft-AI-Chain-Project/mule-ai-chain-connector#34] [General]
<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-00660d27-7fff-b4ec-695b-3a8fcf2d41d8"><ol style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:decimal;font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Need to set multi-release to the correct JDK version, e.g. when using JDK 11, your command should look like this:&nbsp;</span></p></li><ol style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:lower-alpha;font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">mvn clean install -Dmaven.test.skip=true -DskipTests -Djdeps.multiRelease=11</span></p></li></ol><li dir="ltr" style="list-style-type:decimal;font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Need additional permission when using JDK 17 or above, export an environment variable as follows:</span></p></li><ol style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:lower-alpha;font-size:11pt;font-family:Arial,sans-serif;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="2"><p dir="ltr" style="line-height:1.2;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:9pt;font-family:Roboto,sans-serif;color:#1f1f1f;background-color:#ffffff;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">export MAVEN_OPTS=&quot;--add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.regex=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED&quot;</span></p></li></ol></ol><div dir="ltr" style="margin-left:72pt;" align="left">
</div><br /></b>
</body>
</html>

**Comment by amirkhan-ak-sf** [comment]
This is really working! We tried it yesterday on 3 different user machines! Thanks @alick888!

@arpitg-1 @dipeshkumardutta-sfemu FYI.

**Reply by dipeshkumardutta-sfemu** [reply]
As per our connector JDK17 upgrade guide lines we should not use any --add-opens or --add-exports

**Reply by dipeshkumardutta-sfemu** [reply]
Also, why are we compiling on java 17? our mule core modules are not compatible to be compiled using java 17. they can only run on java 17 runtime.

**Reply by amirkhan-ak-sf** [reply]
@dipeshkumardutta-sfemu The point here is to allow the compilation for use cases where Java 8 / 11 can not be installed and used as per policy. It is a workaround that is working until we can officially compile the connectors in Java 17+.

**Reply by dipeshkumardutta-sfemu** [reply]
I do not understand the need of this workaround. I am not aware of which policy. But as of now all our connectors are compiled in JAVA 8 and running on 8, 11 and 17.
So I think the same is applicable for this also.

**Reply by dipeshkumardutta-sfemu** [reply]
Oho! understood now. For those developer who cannot install java 8 in their machine. Am I right @amirkhan-ak-sf ?

**Reply by amirkhan-ak-sf** [reply]
100% yes, thats the use case.
